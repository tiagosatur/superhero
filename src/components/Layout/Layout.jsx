import React from 'react'
import { Container, Grid } from '@material-ui/core'
import { makeStyles } from '@material-ui/styles'

import { Header } from '..'

export default function Layout ({ children }) {
  const classes = useStyles()

  return (
    <Container data-testid='layout'>
      <Grid item xs={12}>
        <Header />
      </Grid>
      <Grid item xs={12} className={classes.main}>
        {children}
      </Grid>
    </Container>
  )
}

const useStyles = makeStyles((theme) => ({
  main: {
    padding: [[64, 0]]
  }
}))
