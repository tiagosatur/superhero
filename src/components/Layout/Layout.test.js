/* eslint-env jest */
import React from 'react'
import '@testing-library/jest-dom/extend-expect'
import { render, cleanup } from '@testing-library/react'

import Layout from './Layout'

describe('Layout component', () => {
  beforeEach(() => {
    cleanup()
  })

  it('matches snapshot', () => {
    const { asFragment } = render(<Layout />)
    expect(
      asFragment(
        <Layout>
          <h2 data-testid='layout-children'>Some child content</h2>
        </Layout>
      )
    ).toMatchSnapshot()
  })

  it('renders ', () => {
    const { queryByTestId } = render(
      <Layout>
        <h2>Some child content</h2>
      </Layout>
    )

    expect(queryByTestId('layout').textContent).not.toEqual(null)
  })

  it('renders its children ', () => {
    const { queryByTestId } = render(
      <Layout>
        <h2>Some child content</h2>
      </Layout>
    )

    expect(queryByTestId('layout')).toBe('layout')
  })
})
