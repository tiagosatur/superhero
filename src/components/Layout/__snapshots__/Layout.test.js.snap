// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiContainer-root MuiContainer-maxWidthLg"
    data-testid="layout"
  >
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
    >
      <div
        class="makeStyles-container-115"
        data-testid="brf-hero-header"
      >
        <header
          class="MuiPaper-root MuiAppBar-root MuiAppBar-positionFixed MuiAppBar-colorPrimary makeStyles-appBar-113 mui-fixed MuiPaper-elevation4"
        >
          <div
            class="MuiToolbar-root MuiToolbar-regular makeStyles-toolbar-116 MuiToolbar-gutters"
          >
            <div
              class="MuiContainer-root makeStyles-materialContainer-117 MuiContainer-maxWidthLg"
            >
              <h2
                class="makeStyles-title-118"
              >
                Super Hero
              </h2>
            </div>
          </div>
        </header>
      </div>
    </div>
    <div
      class="MuiGrid-root makeStyles-main-1 MuiGrid-item MuiGrid-grid-xs-12"
    />
  </div>
</DocumentFragment>
`;
